pykwasm/src/pykwasm/kdist/wasm-semantics/numeric.md

```diff
@@ -136,18 +136,18 @@ Note: The actual `ctz` operator considers the integer 0 to have *all* zero-bits,
     rule ITYPE . ctz    I1 => < ITYPE > #if I1 ==Int 0 #then #width(ITYPE) #else #ctz(I1) #fi
     rule ITYPE . popcnt I1 => < ITYPE > #popcnt(I1)
 
-    syntax Int ::= #minWidth ( Int ) [function]
-                 | #ctz      ( Int ) [function]
-                 | #popcnt   ( Int ) [function]
+    syntax Int ::= #minWidth ( Int ) [function, total]
+                 | #ctz      ( Int ) [function, total]
+                 | #popcnt   ( Int ) [function, total]
  // -------------------------------------------
-    rule #minWidth(0) => 0
-    rule #minWidth(N) => 1 +Int #minWidth(N >>Int 1)                                 requires N =/=Int 0
+    rule #minWidth(N) => 0                                                           requires N <=Int 0
```

Comment:
That will of course create a problem with `clz` again... e.g., 
`int32.clz -1 => #width(i32) -Int #minWidth(-1) => 4 -Int 0 => 4` which is not correct (should be 0 if `-1` is `11...111`)